### DISCUSSION TITLE
MacVim Release 174

### DISCUSSION THREAD

**Discussion by ychin** [macvim-dev/macvim#1289] [Releases]
Updated to Vim 9.0.472.

Announcements
====================

Sponsors
--------------------

MacVim now allows you to sponsor the team! Any amount would be appreciated and it is of course optional. See the announcement (#1271) for more details.

Supporting old versions of macOS (10.12 or below)
--------------------

Currently, macOS supports macOS 10.9 or above. However, this is getting harder to do so as time goes on (e.g. the latest Xcode doesn't support building for it). In the near future, MacVim will mostly target 10.13 (High Sierra) or above, with 10.9 - 10.12 as legacy support. Visit this discussion (#1288) if you would be affected and would like to know more.

Features
====================

Resizing MacVim new options
--------------------

You can now smoothly drag and resize MacVim without having it constrained to fixed multiples of the text size, by setting the "Smoothly resizes window" under General preference pane. This also allows MacVim to behave better when used with third-party tools (e.g. BetterSnapTool) to snap it to one side of the screen. #1276

The command line can now be pinned to the bottom of the window with a setting (under Appearance preference pane). This makes it look better aligned when using smooth resizing, or in other situations where the window size is not a direct multiples of Vim's text size (e.g. full screen or guioption+=k). #1280

Both features in action:

https://user-images.githubusercontent.com/1217449/190357601-1ca6cc82-eb41-4b63-bf0d-55bde73a7607.mp4

New supported text styles
--------------------

Vim highlight groups `strikethrough`, `underdouble`, `underdotted`, `underdashed` are now supported. #1287

<img width="389" alt="MacVim new text styles" src="https://user-images.githubusercontent.com/1217449/190359583-7816d046-a097-4f36-8a0a-ac8530f5131f.png">

New Vim features
--------------------

Virtual text allows you to place custom texts to be displayed inline for diagnostic information and more (e.g. build errors, CSS color preview).  See `:help virtual-text` for more information.

The default color schemes (e.g. desert) are now fixed to look correct again.

Misc options and commands:
- `set nosplitscroll` lets you preserve the windows' scroll positions when splitting. (v9.0.0445)
- `:defer` allows you to clean up in a function. (v9.0.0370)
- `:echowindow` allows you to echo a message to a floating popup message window instead of to the command line. (v9.0.0321)

Fixes
====================

- Fixed non-native full screen not working well with the notch on newer MacBook's when set to not show menu bar. You can also use `MMNonNativeFullScreenSafeAreaBehavior` to force MacVim to use the notch area as well if you don't mind some content being obscured. #1261
- Fixed bad interaction when two settings ("Open untitled window: never" and "After last window closes: Quit MacVim") are set together. #1257
- Fixed the bundled Vim tutor not working when launching from the Help menu #1265.
- Fix crashing on launch under macOS 10.9. #1212
- Fixed potential crash when switching appearance mode in MacVim preferences. #1270
- Korean localized menus no longer throw a syntax error on launch. #1278
- `did_install_default_menus` should work in MacVim now. #1267

General
====================

Sparkle (updater for MacVim) is now updated to 1.27.1 (#1284), and we no longer use DSA keys for signing updates (#1285) as we are using EdDSA already.

Scripting
====================

- Scripting languages versions:
    - Python is now built against 3.10, up from 3.9.

<details>
<summary>Compatibility</summary>

Compatibility
====================

Requires macOS 10.9 or above.

Script interfaces have compatibility with these versions:

- Lua 5.4
- Perl 5.18
- Python2 2.7
- Python3 3.10
- Ruby 3.1
</details>

<hr /><em>This discussion was created from the release <a href='https://github.com/macvim-dev/macvim/releases/tag/release-174'>MacVim Release 174</a>.</em>

**Comment by ychin** [comment]
> Fixed non-native full screen not working well with the notch on newer MacBook's when set to not show menu bar. You can also use `MMNonNativeFullScreenSafeAreaBehavior` to force MacVim to use the notch area as well if you don't mind some content being obscured. #1261

I just found out that this is not working in the built release. This is because we are building the official release using the macOS 11 SDK, *not* macOS 12, which is required for this feature to work. The next release will have this fixed. For now, non-native full screen will still look weird when "Show menu bar in non-native mode" is not set, and `MMNonNativeFullScreenSafeAreaBehavior` won't do anything.

**Comment by dantezhu** [comment]
hi, brew's macvim is not updated to the lastest version.

**Reply by ychin** [reply]
That's managed by Homebrew. Seems like there's already a pull request up: https://github.com/Homebrew/homebrew-core/pull/111060.

It usually does take a couple days for Homebrew to get their formula updated.

**Reply by dantezhu** [reply]
thanks a lot

**Comment by dantezhu** [comment]
<img width="260" alt="image" src="https://user-images.githubusercontent.com/1105212/190959780-b8427a72-2b48-454b-a2e3-a87e3e780c0b.png">
<img width="710" alt="image" src="https://user-images.githubusercontent.com/1105212/190959994-68387082-e1ce-4aaf-a879-10333222b6c6.png">


brew is updated, but it seems requires macOS's version >= 12.6, my macOS'version is 12.5.1

but the README of MacVim is >= 10.9, maybe somewhere cause this problem?

BTW: the MacVim.dmg from github is OK, does not have this problem.

**Reply by ychin** [reply]
Ok I see that. Can you file a bug at Homebrew? As I said, this process is actually controlled by Homebrew. My guess is their tooling somehow set the min SDK requirement to be 12.6. I looked at their https://github.com/Homebrew/homebrew-core/blob/master/Formula/macvim.rb file and didn't see anything wrong there so it could be somewhere else.

But that's also why the official recommendation by us is to just grab the binary release as I have full control over how that is built. The 10.9 support is for the binary releases that we provide at https://github.com/macvim-dev/macvim/releases, and we also support the use case of building source code from scratch. For Homebrew, the min requirement for Homebrew itself is [10.15](https://docs.brew.sh/Installation), so it isn't really possible to go lower than that anyway.

**Reply by dantezhu** [reply]
I see, thanks, I will try to communicate with homebrew.

**Reply by ychin** [reply]
You could also just updated to 12.6. I think it's just a minor update that Apple is doing a phased rollout of. It does seem like Homebrew shouldn't assume you are using the latest software though. They should probably just compile it set as minimum OS 12.0, rather than 12.6.

**Reply by dantezhu** [reply]
Yes, I have upgraded my macOS, It works now. 😂

**Reply by ychin** [reply]
Update (for future reference): Filed an issue at Homebrew (https://github.com/Homebrew/homebrew-core/issues/111693) to track this issue.

**Reply by carlocab** [reply]
Should be fixed now. You can `brew update && brew upgrade macvim` if you haven't managed to upgrade your OS yet.

**Reply by ychin** [reply]
Thanks. Can confirm it works!

**Reply by carlocab** [reply]
Sure; thanks for the report and confirming the fix.

**Comment by igxlin** [comment]
Hi @ychin , look like the tag changes suddenly. It is `snapshot-VERSION` before and `release-VERSION` now. Will we keep using `release-VERSION` in the future?

**Reply by ychin** [reply]
@igxlin Yes I did. Sorry I probably should have said something about this. I have been debating changing it over for a while and slowly migrated the terminology from "snapshot" to "release" in different places as that seemed to make more sense to me. I finally changed the tag format over and was hoping I didn't break anyone's workflow. Can I ask why you care about this? I just want to make sure I didn't break anything.

**Reply by igxlin** [reply]
@ychin 

> Can I ask why you care about this? I just want to make sure I didn't break anything.

Just found that the url changed when I update the [homebrew-cask](https://github.com/Homebrew/homebrew-cask/pull/132249). So I want to check if it is expected. It is OK. No any impact on me.
